examples/voice_send.cr

```diff
@@ -0,0 +1,119 @@
+end
+
+client.on_message_create do |payload|
+  if payload.content.starts_with? "!connect "
+    # Used as:
+    # !connect <guild ID> <channel ID>
+
+    # Parse the command arguments
+    ids = payload.content[9..-1].split(' ').map(&.to_u64)
+
+    client.create_message(payload.channel_id, "Connecting...")
+    connect_channel_id = payload.channel_id
+    client.voice_state_update(ids[0].to_u64, ids[1].to_u64, false, false)
+  elsif payload.content.starts_with? "!play_dca "
+    # Used as:
+    # !play_dca <filename>
+
+    unless voice_client
+      client.create_message(payload.channel_id, "Voice client is nil!")
+      next
+    end
+
+    filename = payload.content[10..-1]
+    file = File.open(filename)
+
+    # The DCAParser class handles parsing of the DCA file. It doesn't do any
+    # sending of audio data to Discord itself – that has to be done by
+    # VoiceClient.
+    parser = Discord::DCAParser.new(file)
+
+    # A proper DCA(1) file contains metadata, which is exposed by DCAParser.
+    # This metadata may be of interest, so here is some example code that uses
+    # it.
+    if parser.metadata
+      metadata = parser.metadata.not_nil!
+      tool = metadata.dca.tool
+      client.create_message(payload.channel_id, "DCA file was created by #{tool.name}, version #{tool.version}.")
+
+      if metadata.info
+        info = metadata.info.not_nil!
+        client.create_message(payload.channel_id, "Song info: #{info.title} by #{info.artist}.") if info.title && info.artist
+      end
+    else
+      client.create_message(payload.channel_id, "DCA file metadata is invalid!")
+    end
+
+    # Set the bot as speaking (green circle). This is important and has to be
+    # done at least once in every voice connection, otherwise the Discord client
+    # will not know who the packets we're sending belongs to.
+    voice_client.not_nil!.send_speaking(true)
+
+    client.create_message(payload.channel_id, "Playing DCA file `#{filename}`.")
+
+    # The `parse` method reads the frames consecutively and parses them to the
+    # block.
+    parser.parse do |frame|
+      # For smooth audio streams Discord requires one packet every
+      # 20 milliseconds. The `timed_run` method measures the time it takes to
+      # run the block and then sleeps 20 milliseconds minus that time, ensuring
+      # accurate timing.
+      #
+      # When simply reading from DCA, the time it takes to read, process and
+      # send the frame is small enough that `timed_run` doesn't make much of a
+      # difference (in fact, some users report that it actually makes things
+      # worse). If the processing time is not negligibly slow because you're
+      # doing something else than DCA parsing, or because you're reading from a
+      # slow source, or for any other reason, then it is recommended to use
+      # `timed_run`. Otherwise, simply doing something like
+      # `sleep 20.milliseconds` may suffice.
+      Discord::VoiceClient.timed_run do
```

Comment:
I added both variants to the example now, so the user can decide by themselves which they want to use.